src/Uno.UI/UI/Xaml/Controls/FlipView/FlipViewAdapter.Android.cs

```diff
@@ -66,6 +66,7 @@ public override void DestroyItem(ViewGroup container, int position, Java.Lang.Ob
 			{
 				//The FlipView doesn't recycle its items on Android, but removing the child view will free up its template to be reused by 
 				// Uno's template pooling. (We don't do this for FlipViewItems explicitly defined in xaml, only for ones generated by us.)
+				(itemView.TemplatedRoot as DependencyObject)?.ClearValue(FrameworkElement.DataContextProperty);
```

Comment:
```suggestion
                // Reset the DataContext from the item, which may have been set by the FlipView. 
                // Since we're manually removing the content, the DataContext may not be reset by 
                // the parent FlipViewItem and will be recycled incorrectly if the DataContext is set.
				(itemView.TemplatedRoot as DependencyObject)?.ClearValue(FrameworkElement.DataContextProperty);

				//The FlipView doesn't recycle its items on Android, but removing the child view will free up its template to be reused by 
				// Uno's template pooling. (We don't do this for FlipViewItems explicitly defined in xaml, only for ones generated by us.)
```